plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/webview/BrowserConnector.kt

```diff
@@ -167,32 +171,59 @@ class BrowserConnector(
                     )
                 )
 
-                val partialResultToken = chatCommunicationManager.addPartialChatMessage(requestFromUi.params.tabId)
                 val chatParams = ChatParams(
                     requestFromUi.params.tabId,
                     chatPrompt,
                     textDocumentIdentifier,
                     cursorState
                 )
 
+                val tabId = requestFromUi.params.tabId
+                val partialResultToken = chatCommunicationManager.addPartialChatMessage(tabId)
+
                 var encryptionManager: JwtEncryptionManager? = null
                 val result = AmazonQLspService.executeIfRunning(project) { server ->
                     encryptionManager = this.encryptionManager
                     encryptionManager?.encrypt(chatParams)?.let { EncryptedChatParams(it, partialResultToken) }?.let { server.sendChatPrompt(it) }
                 } ?: (CompletableFuture.failedFuture(IllegalStateException("LSP Server not running")))
+                showResult(result, partialResultToken, tabId, encryptionManager, browser)
+            }
+            CHAT_QUICK_ACTION -> {
+                val requestFromUi = serializer.deserializeChatMessages(node, QuickChatActionRequest::class.java)
+                val tabId = requestFromUi.params.tabId
+                val quickActionParams = requestFromUi.params
+                val partialResultToken = chatCommunicationManager.addPartialChatMessage(tabId)
+                var encryptionManager: JwtEncryptionManager? = null
+                val result = AmazonQLspService.executeIfRunning(project) { server ->
+                    encryptionManager = this.encryptionManager
```

Comment:
if we are extracting the encryptionManager, we might as well capture the server instance so that we don't need to fetch it on every invocation